### PR TITLE
Types for ThemeContext and BackgroundColorContext

### PR DESCRIPTION
**Pull request by GomezIvann**
Summary:

1. Adding types for these three files.
2. Removing .js extension from every import that referred these files.

### CODE REVIEW

File: lib/src/ThemeContext.tsx [hunk 1]
```diff
@@ -274,14 +274,28 @@ const parseTheme = (theme) => {
...
-  );
+type ThemeProviderCommonProps = {
+  children: React.ReactNode;
+};
+type ThemeProviderDefaultTheme = ThemeProviderCommonProps & {
+  theme?: object;
+  advancedTheme?: never;
+};
+type ThemeProviderAdvancedTheme = ThemeProviderCommonProps & {
+  theme?: never;
```
Comment (marcialfps) [inline lib/src/ThemeContext.tsx:285]: I don't understand the use of `never` type.

Comment (GomezIvann) [reply lib/src/ThemeContext.tsx:285]: `never` is a special type used to represent impossible combinations. In this case, I'm using it to guarantee that an advanced theme is never passed at the same time as a default theme, and the same in the opposite case. For example:

```
type ThemeProviderDefaultTheme = ThemeProviderCommonProps & {
  theme?: object;
  advancedTheme?: never;
};
```

This type represents the situation when the ThemeProvider receives a regular `theme`. By giving `advancedTheme` the type `never` I'm saying that this prop should be received in this case.

Comment (marcialfps) [reply lib/src/ThemeContext.tsx:285]: And why did you discard the first solution? Is it not the same?
```
type ThemeProviderDefaultTheme = ThemeProviderCommonProps & {
  theme: object;
};
```

**Review by marcialfps — APPROVED** [review]

**Comment by marcialfps** [comment]
Please, resolve the conflict if possible. Thanks! ✌🏼

**Comment by GomezIvann** [comment]
> Please, resolve the conflict if possible. Thanks! ✌🏼

Resolved!

**Review by marcialfps — APPROVED** [review]

**Comment by marcialfps** [comment]
We have another conflict lol

**Comment by GomezIvann** [comment]
> We have another conflict lol

Ready again! We need to merge this quickly to avoid conflicts

image

**Review by marcialfps — APPROVED** [review]

**Pull request merged**